cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/WorkflowHttpHandler.java

```diff
@@ -288,11 +277,23 @@ public void getWorkflowSchedules(HttpRequest request, HttpResponder responder,
                                    @PathParam("namespace-id") String namespaceId,
                                    @PathParam("app-id") String appId,
                                    @PathParam("app-version") String appVersion,
-                                   @PathParam("workflow-id") String workflowId) {
+                                   @PathParam("workflow-id") String workflowId) throws NotFoundException {
     ApplicationId application = new ApplicationId(namespaceId, appId, appVersion);
     respondWorkflowSchedules(responder, application, workflowId);
   }
 
+  private void respondWorkflowSchedules(HttpResponder responder, ApplicationId application, String workflow)
+    throws NotFoundException {
+
+    ApplicationSpecification appSpec = store.getApplication(application);
+    if (appSpec == null) {
+      throw new NotFoundException(application);
+    }
+    List<ProgramSchedule> schedules = scheduler.listSchedules(application.workflow(workflow));
+    responder.sendJson(HttpResponseStatus.OK, Schedulers.toScheduleDetails(schedules),
+                       new TypeToken<List<ScheduleDetail>>() { }.getType(), GSON);
```

Comment:
Can we use `SCHEDULE_DETAILS_TYPE`, to avoid this extra object creation each time?